js/app/src/main.ts

```diff
@@ -61,20 +61,26 @@ function create_custom_element(): void {
 
 			await mount_css(ENTRY_CSS, document.head);
 
-			const event = new CustomEvent("domchange", {
+			const domChangeEvent = new CustomEvent("domchange", {
 				bubbles: true,
 				cancelable: false,
 				composed: true
 			});
 
+			const readyEvent = new CustomEvent("ready", {
+				bubbles: true,
+				cancelable: false,
+				composed: true
+			});
+
+
 			const observer = new MutationObserver((mutations) => {
-				this.dispatchEvent(event);
+				this.dispatchEvent(domChangeEvent);
+				this.dispatchEvent(readyEvent);
```

Comment:
We don't want to put this here, this is specific to the DOM changing. We instead want to listen to the `Index` component' `ready` event and dispatch the custom event when it is fired.

Something like 

```js
this.app.$on('ready', () => {
  this.dispatchEvent(...)
)}
  ```